xicam/plugins/__init__.py

```diff
@@ -70,462 +72,351 @@ def load_timer():
+                    except (Exception, SystemError) as ex:
+                        msg.logMessage(
+                            f"Unable to instantiate {entrypoint.name} plugin from module: {entrypoint.module_name}",
+                            msg.ERROR)
+                        msg.logError(ex)
+                        msg.notifyMessage(repr(ex),
+                                          title=f'An error occurred while starting the "{entrypoint.name}" plugin.')
+                    else:
+                        msg.logMessage(f"{int(elapsed() * 1000)} ms elapsed while instantiating {entrypoint.name}",
+                                       level=msg.INFO)
+                        success = True
 
-    def load_element_entry_point(self, category_name, plugin_info):
+                else:
+                    self.type_mapping[type_name][entrypoint.name] = plugin_class
+                    success = True
+
+                if success:
+                    msg.logMessage(f"Successfully collected {entrypoint.name} plugin.", level=msg.INFO)
+                    msg.showProgress(self._progress_count(), maxval=self._entrypoint_count())
+                    self._notify_on_schedule()
+
+            # mark it as completed
+            self._instantiate_queue.task_done()
+
+        # If this was the last plugin
+        if self._load_queue.empty() and self._instantiate_queue.empty() and self.state == State.INSTANTIATING:
+            self.state = State.READY
+            msg.logMessage('Plugin collection completed!')
+            msg.hideProgress()
+
+        if not self.state == State.READY:  # if we haven't reached the last task, but there's nothing queued
+            threads.invoke_as_event(self._instantiate_plugin)  # return to the event loop, but come back soon
+
+    def _get_plugin_by_name(self, name, type_name):
+        return_plugin = None
+        # Check all types matching type_name
+        for search_type_name in self.plugin_types.keys():
+            if type_name == search_type_name or not type_name:
+                match_plugin = self.type_mapping[search_type_name].get(name, None)
+                if match_plugin and return_plugin:
+                    raise NameError('Multiple plugins with the same name but different types exist. '
+                                    'Must specify type_name.')
+                return_plugin = match_plugin
+
+        return return_plugin
+
+    def _get_entrypoint_by_name(self, name, type_name):
+        return_entrypoint = None
+        return_type = None
+        # Check all types matching type_name
+        for search_type_name in self.plugin_types.keys():
+            if type_name == search_type_name or not type_name:
+                match_entrypoint = self._entrypoints[search_type_name].get(name, None)
+                if match_entrypoint and return_entrypoint:
+                    raise NameError('Multiple plugins with the same name but different types exist. '
```

Comment:
The usage of `NameError` here would be confusing to me. From the docs:

> Raised when a local or global name is not found. This applies only to unqualified names. The associated value is an error message that includes the name that could not be found.

Maybe `ValueError` (if the best solution is the provide a `type_name`) or a new custom type `PluginConfigurationError` (if the the best solution is to fix the clashing packages) are worth considering.